deployment/bicep/shared/createvmwindows.bicep

```diff
@@ -0,0 +1,100 @@
+
+param location string = resourceGroup().location
+param subnetId string
+param osDiskType string = 'Standard_LRS'
+param vmSize string = 'Standard_D4_v3'
+param username string
+param password string
+param windowsOSVersion string = '2016-Datacenter'
+
+param vmName string
+param deployAgent bool=false
+
+@description('The Azure DevOps account name]')
+param azureDevOpsAccount string=''
+
+@description('The personal access token to connect to Azure DevOps')
+@secure()
+param personalAccessToken string=''
+
+@description('The Azure DevOps build agent pool for this build agent to join. Use \'Default\' if you don\'t have a separate pool.')
+param poolName string = 'Default'
+
+@description('Enable autologon to run the build agent in interactive mode that can sustain machine reboots.<br>Set this to true if the agents will be used to run UI tests.')
+param enableAutologon bool = false
+
+@description('The base URI where artifacts required by this template are located. When the template is deployed using the accompanying scripts, a private location in the subscription will be used and this value will be automatically generated.')
+param artifactsLocation string = 'https://raw.githubusercontent.com/ahmedsza/azdevopsagent/main/setupagent.ps1'
```

Comment:
This points to a personal github account, fo dev work that's fine but for the final script I'd suggest something different like following the ARm template approach specified in the docs https://docs.microsoft.com/en-us/azure/devops/pipelines/release/deployment-groups/howto-provision-deployment-group-agents?view=azure-devops#use-the-arm-template-deployment-task